### PR TITLE
Make the username in lowercase.

### PR DESCRIPTION
**Pull request by ivanvgh**
### Description:
This PR solves the issue to make sure the username is in lowercase because ICC user creation API only accepts username in lowercase.

### Reviewers:

- [ ] @diegomillan 
- [ ] @Squirrel18

### CODE REVIEW

**Review by diegomillan — APPROVED** [review]

File: openedx_external_enrollments/external_enrollments/icc_external_enrollment.py [hunk 1]
```diff
@@ -186,7 +187,8 @@ def _get_icc_user_from_xml_response(self, response, method_type):
                     icc_user['username'] = field.get('VALUE')
                     break
         except Exception:  # pylint: disable=broad-except
-            pass
+            if response.get('EXCEPTION'):
```
Comment (Squirrel18) [inline openedx_external_enrollments/external_enrollments/icc_external_enrollment.py:190]: Shouldn't it log the exception regardless of the response? @ivanvgh

Comment (ivanvgh) [reply openedx_external_enrollments/external_enrollments/icc_external_enrollment.py:190]: It is not necessary at this point but I can be helpful. Already added.

File: openedx_external_enrollments/external_enrollments/icc_external_enrollment.py [hunk 2]
```diff
@@ -185,8 +186,15 @@ def _get_icc_user_from_xml_response(self, response, method_type):
                 if field.get('@name') == 'username':
                     icc_user['username'] = field.get('VALUE')
                     break
-        except Exception:  # pylint: disable=broad-except
-            pass
+        except (TypeError, KeyError):  # pylint: disable=broad-except
+            pass # User not found.
```
Comment (diegomillan) [inline openedx_external_enrollments/external_enrollments/icc_external_enrollment.py:190]: We should log this event, at least using python `LOG.warning()`

**Review by diegomillan — CHANGES_REQUESTED** [review]

**Review by diegomillan — APPROVED** [review]

File: openedx_external_enrollments/external_enrollments/icc_external_enrollment.py (hunk 1 above)
Comment (Squirrel18) [reply openedx_external_enrollments/external_enrollments/icc_external_enrollment.py]: Great, thanks.

File: openedx_external_enrollments/external_enrollments/icc_external_enrollment.py [hunk 3]
```diff
@@ -185,8 +186,15 @@ def _get_icc_user_from_xml_response(self, response, method_type):
                 if field.get('@name') == 'username':
                     icc_user['username'] = field.get('VALUE')
                     break
-        except Exception:  # pylint: disable=broad-except
-            pass
+        except (TypeError, KeyError):  # pylint: disable=broad-except
```
Comment (Squirrel18) [inline openedx_external_enrollments/external_enrollments/icc_external_enrollment.py:189]: Please, move the broad-except flag to the correct exception. @ivanvgh

Comment (ivanvgh) [reply openedx_external_enrollments/external_enrollments/icc_external_enrollment.py]: You are right, Done.

**Review by Squirrel18 — APPROVED** [review]

**Pull request merged**